docs/src/main/tut/datatypes/eithert.md

```diff
@@ -0,0 +1,160 @@
+---
+layout: docs
+title:  "EitherT"
+section: "data"
+source: "core/src/main/scala/cats/data/EitherT.scala"
+scaladoc: "#cats.data.EitherT"
+---
+# EitherT
+
+`Either` can be used for error handling in most situations. However, when
+`Either` is placed into effectful types such as `Option` or`Future`, a large
+amount of boilerplate is required to handle errors. For example, consider the
+following program:
+
+```tut:book
+import scala.util.Try
+import cats.syntax.either._
+
+def parseDouble(s: String): Either[String, Double] =
+  Try(s.toDouble).map(Right(_)).getOrElse(Left(s"$s is not a number"))
+
+def divide(a: Double, b: Double): Either[String, Double] =
+  Either.cond(b != 0, a / b, "Cannot divide by zero")
+
+def divisionProgram(inputA: String, inputB: String): Either[String, Double] =
+  for {
+    a <- parseDouble(inputA)
+    b <- parseDouble(inputB)
+    result <- divide(a, b)
+  } yield result
+
+divisionProgram("4", "2") // Right(2.0)
+divisionProgram("a", "b") // Left("a is not a number")
+```
+
+Suppose `parseDouble` and `divide` are rewritten to be asynchronous and return
+`Future[Either[String, Double]]` instead. The for-comprehension can no longer be
+used since `divisionProgram` must now compose `Future` and `Either` together,
+which means that the error handling must be performed explicitly to ensure that
+the proper types are returned:
+
+```tut:silent
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+def parseDoubleAsync(s: String): Future[Either[String, Double]] =
+  Future.successful(parseDouble(s))
+def divideAsync(a: Double, b: Double): Future[Either[String, Double]] =
+  Future.successful(divide(a, b))
+
+def divisionProgramAsync(inputA: String, inputB: String): Future[Either[String, Double]] =
+  parseDoubleAsync(inputA) flatMap { eitherA =>
+    parseDoubleAsync(inputB) flatMap { eitherB =>
+      val parseResult = for {
+        a <- eitherA
+        b <- eitherB
+      } yield (a, b)
```

Comment:
You could write `(eitherA, eitherB).tupled`.

Reply:
That syntax seems obscure to me, so I'm just replacing the `parseResult` snippet with a pattern match on the eithers.